core/regexp/timeout_spec.rb

```diff
@@ -0,0 +1,42 @@
+require_relative '../../spec_helper'
+
+ruby_version_is "3.2" do
+  describe "Regexp.timeout" do
+    after :each do
+      Regexp.timeout = nil
+    end
+
+    it "returns global timeout" do
+      Regexp.timeout = 3
+      Regexp.timeout.should == 3
+    end
+
+    it "raises Regexp::TimeoutError after global timeout elapsed" do
+      Regexp.timeout = 0.001
+      Regexp.timeout.should == 0.001
+
+      t = Time.now
+      -> {
+        # A typical ReDoS case
+        /^(a*)*$/ =~ "a" * 1000000 + "x"
+      }.should raise_error(Regexp::TimeoutError, "regexp match timeout")
+      t = Time.now - t
+      t.should be_close(0.001, 0.010)
```

Comment:
This is likely too fragile, the OS could decide to pause this process just after the exception because it's overloaded and then it would take longer in clock time.
I think we need to drop these Time checks and only use the `should raise_error`.
If a spec takes 3 seconds anyway we'll notice.

Reply:
Sure @eregon I'll remove Time checks